lang/java/reef-wake/wake/src/main/java/org/apache/reef/wake/remote/transport/netty/NettyHttpServerEventListener.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.reef.wake.remote.transport.netty;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.channel.Channel;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.handler.codec.DecoderResult;
+import io.netty.handler.codec.http.*;
+import io.netty.util.CharsetUtil;
+import org.apache.reef.wake.EStage;
+import org.apache.reef.wake.remote.impl.ByteCodec;
+import org.apache.reef.wake.remote.impl.TransportEvent;
+
+import java.net.SocketAddress;
+import java.net.URI;
+import java.util.Map;
+import java.util.concurrent.ConcurrentMap;
+import java.util.logging.Level;
+
+/**
+ * A Netty event listener for server side.
+ */
+final class NettyHttpServerEventListener extends AbstractNettyEventListener {
+
+  private HttpRequest httpRequest;
+  private final StringBuilder buf = new StringBuilder();
+  private final URI uri;
+
+  NettyHttpServerEventListener(
+      final ConcurrentMap<SocketAddress, LinkReference> addrToLinkRefMap,
+      final EStage<TransportEvent> stage,
+      final URI uri) {
+    super(addrToLinkRefMap, stage);
+    this.uri = uri;
+  }
+
+
+  @Override
+  public void channelActive(final ChannelHandlerContext ctx) {
+    final Channel channel = ctx.channel();
+
+    if (LOG.isLoggable(Level.FINEST)) {
+      LOG.log(Level.FINEST, "Channel active. key: {0}", channel.remoteAddress());
+    }
+
+    this.addrToLinkRefMap.putIfAbsent(
+        channel.remoteAddress(), new LinkReference(new NettyLink<>(
+            channel, new ByteCodec(), new LoggingLinkListener<byte[]>(), uri)));
+
+    LOG.log(Level.FINER, "Add connected channel ref: {0}", this.addrToLinkRefMap.get(channel.remoteAddress()));
+
+  }
+
+  @Override
+  public void channelRead(final ChannelHandlerContext ctx, final Object msg) {
+    final HttpHeaders headers;
+    final HttpRequest request;
```

Comment:
As only one of these two variables will be used, I think it would be better to define each variable inside each `if` and `else if`.